src/modules/acbs/acbs-party.service.create-party.test.ts

```diff
@@ -0,0 +1,135 @@
+import { HttpService } from '@nestjs/axios';
+import { CreatePartyGenerator } from '@ukef-test/support/generator/create-party-generator';
+import { RandomValueGenerator } from '@ukef-test/support/generator/random-value-generator';
+import { AxiosError } from 'axios';
+import { when } from 'jest-when';
+import { of, throwError } from 'rxjs';
+
+import { DateStringTransformations } from '../date/date-string.transformations';
+import { AcbsPartyService } from './acbs-party.service';
+import { AcbsBadRequestException } from './exception/acbs-bad-request.exception';
+import { AcbsUnexpectedException } from './exception/acbs-unexpected.exception';
+
+describe('AcbsPartyService', () => {
+  const valueGenerator = new RandomValueGenerator();
+  const dateStringTransformations = new DateStringTransformations();
+  const idToken = valueGenerator.string();
+  const baseUrl = valueGenerator.httpsUrl();
+
+  let httpService: HttpService;
+  let service: AcbsPartyService;
+
+  let httpServicePost: jest.Mock;
+
+  const { acbsCreatePartyRequest: newParty } = new CreatePartyGenerator(valueGenerator, dateStringTransformations).generate({
+    numberToGenerate: 1,
+  });
+
+  const expectedHttpServicePostArgs = [
+    `/Party`,
+    newParty,
+    {
+      baseURL: baseUrl,
+      headers: { Authorization: `Bearer ${idToken}`, 'Content-Type': 'application/json' },
+    },
+  ];
+
+  beforeEach(() => {
+    httpService = new HttpService();
+
+    httpServicePost = jest.fn();
+    httpService.post = httpServicePost;
+
+    service = new AcbsPartyService({ baseUrl }, httpService);
+  });
+
+  describe('createParty', () => {
+    it('sends a POST to ACBS to create a party with the specified parameters', async () => {
```

Comment:
Can we possibly add few extra assertions regarding party ID with void values:
```
            headers: {
              location: `/Party/[Input],
            },
```

* <String>
* ABCD
* !"£!"
* `  `